lib/conjur/id.rb

```diff
@@ -52,7 +55,9 @@ def as_json options={}
 
     # Splits the id into 3 components, and then joins them with a forward-slash `/`.
     def to_url_path
-      id.split(':', 3).join('/')
+      id.split(':', 3)
+        .map { |part| self.class.path_escape(part) }
```

Comment:
And, once you've included `Conjur::Escape::ClassMethods`, you can write this as `.map(&:path_escape)`, instead.